### PR TITLE
keytree: new crate

### PR DESCRIPTION
**Pull request by tessr**
This adds a new crate `keytree` and provides
a function for generating a new private key. 

Thank you @cathieyun for guiding me through this!

### CODE REVIEW

File: keytree/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+use curve25519_dalek::scalar::Scalar;
```
Comment (oleganza) [inline keytree/src/lib.rs]: let's add `#![deny(missing_docs)]` to the top of the lib.rs

File: keytree/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/target
+**/*.rs.bk
+Cargo.lock
```
Comment (oleganza) [inline keytree/.gitignore]: trailing newline to make git happy?

File: keytree/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+authors = ["Tess Rinearson <tess.rinearson@gmail.com>"]
+edition = "2018"
+
+[dependencies]
+curve25519-dalek = { version = "1.0.1", features = ["serde"] }
+merlin = "1.0.1"
+rand = "0.6"
+
+[dev-dependencies]
+rand_chacha = "0.1"
```
Comment (oleganza) [inline keytree/Cargo.toml]: trailing newline to make git happy?

**Comment by oleganza** [comment]
Let's add this to `.travis.yml` too:

```yml
  - language: rust
    rust: nightly-2018-12-31
    # per https://levans.fr/rust_travis_cache.html
    cache:
      directories:
        - /home/travis/.cargo
    before_cache:
      - rm -rf /home/travis/.cargo/registry
    before_script:
    - cd keytree
    - rustup component add rustfmt-preview
    script:
    - cargo fmt --all -- --check
    - cargo test
    - RUSTFLAGS="-C opt-level=0" cargo bench "DONOTMATCHANYBENCHMARK"
```

**Comment by oleganza** [comment]
Notice, that this travis rule above has `cargo fmt` in it - you may want to run it before pushing to avoid waiting a few minutes only to find out that a whitespace is slightly off.

**Comment by tessr** [comment]
PTAL :cat_with_crab_eyes:

**Pull request merged**